pkg/apps/packagekit.js

```diff
@@ -114,66 +81,46 @@ export function refresh(origin_files, config_packages, data_packages, progress_c
     const progress = new ProgressReporter(0, 1, progress_cb);
 
     const search_origin_file_packages = () => {
-        return PK.cancellableTransaction("SearchFiles", [PK.Enum.FILTER_INSTALLED, origin_files],
-                                         progress.progress_reporter,
-                                         {
-                                             Package: (info, package_id) => {
-                                                 const pkg = package_id.split(";")[0];
-                                                 origin_pkgs[pkg] = true;
-                                             },
-                                         });
+        return PK.find_file_packages(origin_files, progress.progress_reporter).then(packages => {
+            for (const pkgname of packages) {
+                origin_pkgs[pkgname] = true;
+            }
+        });
     };
 
     const refresh_cache = () => {
         progress.base = 6;
         progress.range = 69;
 
-        return PK.cancellableTransaction("RefreshCache", [true], progress.progress_reporter);
+        return PK.refresh(true, progress.progress_reporter);
     };
 
     const maybe_update_origin_file_packages = () => {
         progress.base = 75;
         progress.range = 5;
 
-        return PK.cancellableTransaction("GetUpdates", [0], progress.progress_reporter,
-                                         {
-                                             Package: (info, package_id) => {
-                                                 const pkg = package_id.split(";")[0];
-                                                 if (pkg in origin_pkgs)
-                                                     update_ids.push(package_id);
-                                             },
-                                         })
-                .then(() => {
-                    progress.base = 80;
-                    progress.range = 15;
-
-                    if (update_ids.length > 0)
-                        return PK.cancellableTransaction("UpdatePackages", [0, update_ids],
-                                                         progress.progress_reporter);
-                });
+        return PK.get_updates(false, progress.progress_reporter).then(updates => {
+            const filtered_updates = [];
+
+            for (const update of updates) {
+                if (update.name in origin_pkgs)
+                    filtered_updates.push(update);
+            }
+
+            progress.base = 80;
+            progress.range = 15;
+
+            if (filtered_updates.length > 0)
+                return PK.update_packages(filtered_updates, progress.progress_reporter, null);
```

Comment:
This added line is not executed by any test.  [Details](https://logs-cockpit.apps.ocp.cloud.ci.centos.org/pull-22498-af4148aa-20251119-125328-fedora-43-devel/Coverage/lcov/github-pr.diff.gcov.html)